Framework/Core/src/Dispatcher.cxx

```diff
@@ -48,9 +49,16 @@ void Dispatcher::init(InitContext& ctx)
 {
   LOG(DEBUG) << "Reading Data Sampling Policies...";
 
-  std::unique_ptr<ConfigurationInterface> cfg = ConfigurationFactory::getConfiguration(mReconfigurationSource);
-  auto policiesTree = cfg->getRecursive("dataSamplingPolicies");
-  mPolicies.clear();
+  boost::property_tree::ptree policiesTree;
+
+  if (mReconfigurationSource.empty() == false) {
+    std::unique_ptr<ConfigurationInterface> cfg = ConfigurationFactory::getConfiguration(mReconfigurationSource);
+    policiesTree = cfg->getRecursive("dataSamplingPolicies");
```

Comment:
This comes from the old code. I would actually retrieve the `mReconfigurationSource` at each invokation, since it might have been changed across run.

Reply:
That would be a risky thing to do during the run, but indeed it could be retrieved each time. I can do that with a bunch of other planned improvements.